cy.mount cannot mount multiple components

Using cy.mount to mount one component can be finished successfully but failed to mount multiple components.

Please find the files in attached zip file (component.and.test.files.zip).
*Button.js* is the file of component. *button.spec.js* is the test file.
[component and test files.zip](https://github.com/bahmutov/cypress-react-unit-test/files/3312909/component.and.test.files.zip)
snapshot of error info
![image](https://user-images.githubusercontent.com/29821872/59896097-4ac64580-943b-11e9-90f7-034aee9fdcf7.png)
error log
```

cypress_runner.js:172385 Error:    TypeError: Invalid value used as weak map key
    at WeakMap.set (<anonymous>)
    at Object.getDisplayName [as default] (http://localhost:51524/__cypress/tests?p=cypress/integration/button.spec.js-500:254:24)
    at Context.exports.mount (http://localhost:51524/__cypress/tests?p=cypress/integration/button.spec.js-500:356:47)
From previous event:
    at runCommand (http://localhost:51524/__cypress/runner/cypress_runner.js:78709:14)
    at next (http://localhost:51524/__cypress/runner/cypress_runner.js:78793:14)
From previous event:
    at http://localhost:51524/__cypress/runner/cypress_runner.js:78812:37
From previous event:
    at run (http://localhost:51524/__cypress/runner/cypress_runner.js:78810:15)
    at Object.cy.<computed> [as mount] (http://localhost:51524/__cypress/runner/cypress_runner.js:79038:11)
    at Context.runnable.fn (http://localhost:51524/__cypress/runner/cypress_runner.js:79168:20)
    at callFn (http://localhost:51524/__cypress/runner/cypress_runner.js:51457:21)
    at Test.Runnable.run (http://localhost:51524/__cypress/runner/cypress_runner.js:51450:7)
    at http://localhost:51524/__cypress/runner/cypress_runner.js:82159:28
From previous event:
    at Object.onRunnableRun (http://localhost:51524/__cypress/runner/cypress_runner.js:82154:20)
    at $Cypress.action (http://localhost:51524/__cypress/runner/cypress_runner.js:77456:51)
    at Test.Runnable.run (http://localhost:51524/__cypress/runner/cypress_runner.js:81220:20)
    at Runner.runTest (http://localhost:51524/__cypress/runner/cypress_runner.js:51920:10)
    at http://localhost:51524/__cypress/runner/cypress_runner.js:52026:12
    at next (http://localhost:51524/__cypress/runner/cypress_runner.js:51840:14)
    at http://localhost:51524/__cypress/runner/cypress_runner.js:51850:7
    at next (http://localhost:51524/__cypress/runner/cypress_runner.js:51782:14)
    at http://localhost:51524/__cypress/runner/cypress_runner.js:51818:5
    at timeslice (http://localhost:51524/__cypress/runner/cypress_runner.js:47059:27)
```

--- a/cypress/component/advanced/lazy-loaded/lazy-load-spec.js
+++ b/cypress/component/advanced/lazy-loaded/lazy-load-spec.js
@@ -1,4 +1,6 @@
 // https://github.com/bahmutov/cypress-react-unit-test/issues/136
+// dynamic imports like this work in example projects, but not inside this repo
+// probably due to webpack plugins not set up correctly ☹️
 describe.skip('dynamic import', () => {
   it('loads', () => {
     // cy.wrap(import('./lazy-add'))

--- a/cypress/component/advanced/material-ui-example/list-item-spec.js
+++ b/cypress/component/advanced/material-ui-example/list-item-spec.js
@@ -6,12 +6,11 @@ import { ListItemText } from '@material-ui/core'
 import SimpleList from './list-demo'
 
 it('renders a list item', () => {
-  const Demo = () => (
+  mount(
     <ListItem>
       <ListItemText primary={'my example list item'} />
-    </ListItem>
+    </ListItem>,
   )
-  mount(<Demo />)
   cy.contains('my example list item')
 })
 

--- a/cypress/component/advanced/portal/portal-spec.js
+++ b/cypress/component/advanced/portal/portal-spec.js
@@ -6,18 +6,16 @@ const RenderInPortal = ({ children }) => {
   return ReactDOM.createPortal(children, document.body)
 }
 
-const MyComponent = () => (
-  <div id="component">
-    <p>Hello World!</p>
-    <RenderInPortal>
-      <p> I am in portal </p>
-    </RenderInPortal>
-  </div>
-)
-
 describe('Portals', () => {
   it('renders content inside the portal', () => {
-    mount(<MyComponent />)
+    mount(
+      <div id="component">
+        <p>Hello World!</p>
+        <RenderInPortal>
+          <p> I am in portal </p>
+        </RenderInPortal>
+      </div>,
+    )
 
     cy.contains('#component', 'Hello World!')
     cy.get('#component').should('not.contain', 'I am in portal')

--- a/cypress/component/basic/mount-div/spec.js
+++ b/cypress/component/basic/mount-div/spec.js
@@ -1,0 +1,27 @@
+/// <reference types="cypress" />
+/// <reference types="../../lib" />
+import React from 'react'
+import { mount } from 'cypress-react-unit-test'
+
+function Button() {
+  return <button>Hello</button>
+}
+
+describe('mounting a div', () => {
+  it('works', () => {
+    mount(<div className="example">Works</div>)
+    cy.contains('Works').should('be.visible')
+  })
+
+  // https://github.com/bahmutov/cypress-react-unit-test/issues/98
+  it('mount multiple components', function() {
+    mount(
+      <div>
+        <Button />
+        <hr />
+        <Button />
+      </div>,
+    )
+    cy.get('button').should('have.length', 2)
+  })
+})

--- a/lib/getDisplayName.ts
+++ b/lib/getDisplayName.ts
@@ -49,7 +49,9 @@ export default function getDisplayName(
     }
   }
 
-  cachedDisplayNames.set(type, displayName)
+  try {
+    cachedDisplayNames.set(type, displayName)
+  } catch (e) {}
 
   return displayName
 }

--- a/lib/index.ts
+++ b/lib/index.ts
@@ -45,7 +45,7 @@ export const mount = (jsx: React.ReactElement, options: MountOptions = {}) => {
 
   // Get the display name property via the component constructor
   // @ts-ignore FIXME
-  const displayname = getDisplayName(jsx.type, options.alias)
+  const displayName = getDisplayName(jsx.type, options.alias)
   let logInstance: Cypress.Log
 
   return cy
@@ -53,7 +53,7 @@ export const mount = (jsx: React.ReactElement, options: MountOptions = {}) => {
       if (options.log !== false) {
         logInstance = Cypress.log({
           name: 'mount',
-          message: [`ReactDOM.render(<${displayname} ... />)`],
+          message: [`ReactDOM.render(<${displayName} ... />)`],
         })
       }
     })
@@ -99,7 +99,7 @@ export const mount = (jsx: React.ReactElement, options: MountOptions = {}) => {
 
       return cy
         .wrap(CypressTestComponent, { log: false })
-        .as(options.alias || displayname)
+        .as(options.alias || displayName)
         .then(() => {
           if (logInstance) {
             logInstance.snapshot('mounted')

